patches/server/0609-Add-Unix-domain-socket-support.patch

```diff
@@ -139,3 +140,42 @@ index 50c79be12cf8778a2ce451a8c9bb2cdc277d6830..8393c1a2d15d5ad255c9808b0d0edd6a
                              connection.spoofedUUID = com.mojang.util.UUIDTypeAdapter.fromString( split[2] );
                          } else
                          {
+diff --git a/src/main/java/net/minecraft/server/network/ServerLoginPacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
+index b026c003e1fc02a9ea426f3126acb788fc09a874..39b4d82f8f791bc2b6a717209c651f04f7a0da2b 100644
+--- a/src/main/java/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
++++ b/src/main/java/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
+@@ -331,7 +331,13 @@ public class ServerLoginPacketListenerImpl implements ServerLoginPacketListener,
+         public void fireEvents() throws Exception {
+                         String playerName = ServerLoginPacketListenerImpl.this.gameProfile.getName();
+                         java.net.InetAddress address = ((java.net.InetSocketAddress) ServerLoginPacketListenerImpl.this.connection.getRemoteAddress()).getAddress();
+-                        java.net.InetAddress rawAddress = ((java.net.InetSocketAddress) connection.channel.remoteAddress()).getAddress(); // Paper
++                        // Paper start
++                        java.net.SocketAddress rawSocketAddress = connection.channel.remoteAddress();
++                        java.net.InetAddress rawAddress = java.net.InetAddress.getLoopbackAddress();
++                        if (rawSocketAddress != null && rawSocketAddress instanceof java.net.InetSocketAddress rawInetSocketAddress) {
```

Comment:
Null check here can be removed here and in the other spot, the instanceof takes care of it